replaced python codes to github links

youngsun4786 opened the pull request.

itsjoeoui commented on src/feature/flow_chart_panel/views/NodeModal.tsx:
```diff
@@ -17,9 +19,36 @@ const NodeModal = ({
+      LINK = `${GLINK}/${nodeCategory}/${nodeTypeLink}/${pythonString}`;
+      break;
+    case "OBJECT_DETECTION":
+      LINK = `${GLINK}/AI_ML/${nodeDataLabel}/${pythonString}`;
```
We should probably not hardcode "AL_ML" here I think.

youngsun4786 replied:
Oh okay! The file name was acquired using the title property from COMMANDS_MANIFEST.ts file and the title name was something like "AI and Machine Learning". What alternative do you suggest in this situation?
![Screenshot 2023-05-23 at 2 59 16 PM](https://github.com/flojoy-io/studio/assets/83793112/fcb1a75f-8c71-4bd0-9dc8-089599f53533)

youngsun4786 replied:
I agree with hard coding being a malpractice here.

itsjoeoui replied:
@smahmed776 Shall we just change the title to stay consistent in this case?

Jin1L replied:
@itsjoeoui Similar thing for CLOUD_DATABASE

smahmed776 replied:
@itsjoeoui  We can add a key field for each parent category and update all folders not matching the key(if any) and keep the title as it is because it looks better with spacing and full name in the sidebar than these shorthands IMO. Also, we need to ensure each node folder follows its file name like `LINSPACE/LINSPACE.py`. This should resolve all the issues.

note: We need to update docs in the docs repo to force app builders to follow this convention.

itsjoeoui replied:
Sounds good! I think we can start with that such that we can have a better structure, I can take a look in a bit

youngsun4786 replied:
Sounds good! I'll be working on it today!

youngsun4786 commented:
> 



> 

@smahmed776 Advised me to start working on the UX for the nodes along with this PR, hence the progress so far looks like this if you were referring to the UX:
![Screenshot 2023-05-24 at 5 00 04 PM](https://github.com/flojoy-io/studio/assets/83793112/0659eecc-ec36-4c81-a7f6-2b6fd76a6639)

The links do work except I still have to change the structure for manifest a bit and rename some files.

jackparmer commented:
can you add some screenshots please @youngsun4786 ?

youngsun4786 commented:
@jackparmer Hey Jack! I apologize for a late response. 
I've noticed that on there was a secondary button color on https://www.flojoy.io/brand so I made the corresponding changes.
- modified the logic for getting file paths for the github link.
- added the colors to code examples & incorporated `JSONTree` for jsons code.

https://github.com/flojoy-io/studio/assets/83793112/b31de4d3-51a7-49c4-9ec3-c5ae605ae821

smahmed776 commented on src/feature/flow_chart_panel/FlowChartTabView.tsx:
```diff
@@ -205,16 +207,18 @@ const FlowChartTab = () => {
 
     if (selectedNode.data.func === "CONSTANT") {
-      pythonString = PYTHON_FUNCTIONS[selectedNode.data.func + ".py"];
+      nodeFileName = selectedNode.data.func + ".py";
```
This line applies for all the nodes not only `CONSTANT`

smahmed776 commented on src/feature/flow_chart_panel/views/NodeModal.tsx:
```diff
@@ -18,30 +189,111 @@ const NodeModal = ({
+    }
+  }
+
+  switch (nodeCategory) {
```
Can we set a name convention for all categories/sub-categories/files so that we can get rid of this `switch` method and we don't have to change here again in case of new category/file is introduced?

smahmed776 commented on src/feature/flow_chart_panel/views/NodeModal.tsx:
```diff
@@ -61,22 +313,27 @@ const NodeModal = ({
       <SyntaxHighlighter
         language="python"
-        style={colorScheme === "dark" ? srcery : docco}
+        style={colorScheme === "dark" ? darkFlojoy : lightFlojoy}
```
This line gives a type error to me
